src/data/auroradb.ts

```diff
@@ -0,0 +1,251 @@
+            indicator: item.indicator,
+            street: item.street,
+        }));
+    } catch (error) {
+        console.error(`Error performing batch get for naptan info for stop list: ${atcoCodes}, error: ${error}`);
+        throw new Error(error);
+    }
+};
+
+export const getAtcoCodesByNaptanCodes = async (naptanCodes: string[]): Promise<NaptanAtcoCodes[]> => {
+    const listOfNaptanCodes = naptanCodes.toString();
+    const atcoCodesByNaptanCodeQuery = `SELECT * FROM naptanStop WHERE naptanCode IN ?`;
+
+    try {
+        const results = await executeQuery(atcoCodesByNaptanCodeQuery, [listOfNaptanCodes]);
+
+        const queryResults = JSON.parse(JSON.stringify(results));
+        return queryResults.map((item: any) => ({ atcoCode: item.atcoCode, naptanCode: item.NaptanCode }));
+    } catch (error) {
+        console.error(
+            `Error performing queries for ATCO Codes using Naptan Codes: ${naptanCodes}. Error: ${error.stack}`,
+        );
+        throw new Error(error);
+    }
+};
+
+export const getServiceByNocCodeAndLineName = async (nocCode: string, lineName: string): Promise<RawService> => {
+    const serviceQuery =
+        'SELECT os.operatorShortName, os.serviceDescription, os.lineName, pl.fromAtcoCode, pl.toAtcoCode, pl.journeyPatternId, pl.orderInSequence, nsStart.commonName AS fromCommonName, nsStop.commonName as toCommonName ' +
+        ' FROM tndsOperatorService AS os' +
+        ' INNER JOIN tndsJourneyPattern AS ps ON ps.operatorServiceId = os.id' +
+        ' INNER JOIN tndsJourneyPatternLink AS pl ON pl.journeyPatternId = ps.id' +
+        ' LEFT JOIN naptanStop nsStart ON nsStart.atcoCode=pl.fromAtcoCode' +
+        ' LEFT JOIN naptanStop nsStop ON nsStop.atcoCode=pl.toAtcoCode' +
+        ' WHERE os.nocCode = ? AND os.lineName = ?' +
+        ' ORDER BY pl.orderInSequence, pl.journeyPatternId ASC';
+
+    let queryItems: QueryData[];
+
+    try {
+        const queryResult = await executeQuery(serviceQuery, [nocCode, lineName]);
+        queryItems = JSON.parse(JSON.stringify(queryResult));
+    } catch (err) {
+        throw new Error(`Could not get journey patterns from Aurora DB: ${err.name}, ${err.message}`);
+    }
+
+    const service = queryItems?.[0];
+    const rawPatternService: RawJourneyPattern[] = [];
+
+    // allows to get the unique journey's for the operator e.g. [1,2,3]
+    const uniqueJourneyPatterns = queryItems
+        .map(item => item.journeyPatternSectionId)
+        .filter((value, index, self) => self.indexOf(value) === index);
+
+    uniqueJourneyPatterns.forEach(journey => {
+        const filteredJourney = queryItems.filter(item => {
+            return item.journeyPatternSectionId === journey;
+        });
+
+        const journeyPatternSection: RawJourneyPatternSection = {
+            OrderedStopPoints: filteredJourney.map((data: QueryData) => {
+                return {
+                    StopPointRef: data.toAtcoCode,
```

Comment:
This is leaving out the first stop in the journey pattern since we're only using the `toAtcoCode` and `toCommonName`, for the first link we'll also need the `fromAtcoCode` and `fromCommonName`, maybe something like:

```
 OrderedStopPoints: [
    {
        StopPointRef: filteredJourney[0].fromAtcoCode,
        CommonName: filteredJourney[0].fromCommonName,
    },
    ...filteredJourney.map((data: QueryData) => ({
        StopPointRef: data.toAtcoCode,
        CommonName: data.toCommonName,
    })),
],
```